src/System.Management.Automation/FormatAndOutput/common/ComplexWriter.cs

```diff
@@ -353,27 +355,62 @@ static StringManipulationHelper()
         private static IEnumerable<GetWordsResult> GetWords(string s)
         {
             StringBuilder sb = new StringBuilder();
-            GetWordsResult result = new GetWordsResult();
+            StringBuilder vtSeqs = null;
+            Dictionary<int, int> vtRanges = null;
 
+            var valueStrDec = new ValueStringDecorated(s);
+            if (valueStrDec.IsDecorated)
+            {
+                vtSeqs = new StringBuilder();
+                vtRanges = new Dictionary<int, int>();
+                foreach (Match match in ValueStringDecorated.AnsiRegex.Matches(s))
+                {
+                    vtRanges.Add(match.Index, match.Length);
+                }
+            }
+
+            bool wordHasVtSeqs = false;
             for (int i = 0; i < s.Length; i++)
             {
-                // Soft hyphen = \u00AD - Should break, and add a hyphen if needed. If not needed for a break, hyphen should be absent
-                if (s[i] == ' ' || s[i] == '\t' || s[i] == s_softHyphen)
+                if (vtRanges?.TryGetValue(i, out int len) == true)
                 {
-                    result.Word = sb.ToString();
-                    sb.Clear();
-                    result.Delim = new string(s[i], 1);
+                    var vtSpan = s.AsSpan(i, len);
+                    sb.Append(vtSpan);
+                    vtSeqs.Append(vtSpan);
 
-                    yield return result;
+                    wordHasVtSeqs = true;
+                    i += len - 1;
+                    continue;
```

Comment:
Why do we skip vt sequence if the line ends with delimiter and vt sequence?

Reply:
Because in that case the last word/line will consist of VT sequences only, with no actual character to be rendered. To the formatting system, it's the same as an empty string, so we just skip to avoid unneeded string allocations.

Reply:
You think about formatting codes only but there are non-formatting codes too like OSC 2 - set windows title.

Reply:
What is OSC 2? And how does that go through the formatting code here? Can you show a couple examples where this is a problem?

Reply:
OSC 2 is "set windows title". I want to say VT sequences are not only formatting but control codes too.

Reply:
For the "set title" control sequence in particular, the syntax is: [`ESC]2;stringBEL`](https://tldp.org/HOWTO/Xterm-Title-3.html), so there is no point to keep the VT sequence if `stringBEL` part doesn't exist.

In general, you shouldn't expect control sequences to work with PowerShell formatting. Formatting by definition is just for visualizing objects, and in my opinion, accidentally making a control sequence take effect is a bad side effect of the formatting, but that's inevitable today given that our `AnsiRegex` cannot tell VT sequences for visual effects from other VT sequences.